### PR TITLE
Documentation, pragmas

### LINKED ISSUE #9454: documentation of {.noRewrite.} pragma is missing
While looking into term-rewriting macros recursion, I realised that the one true fix to #9288 #8376 and #2901 mentioned in the doc PR #9305 was introduced in 2015, the forgotten {.noRewrite.} pragma https://github.com/nim-lang/Nim/commit/4674b3eb8003b44656ae15ff0a5b7fc742714597.

It's completely undocumented though and I'm not even sure @Araq was aware of it:

> Introduce {.noRewrite.} expr pragma disabling TR
> 
> Term rewriting macros/templates are currently greedy and they will rewrite as
> long as there is a match. So there was no way to ensure some rewrite happens
> only once, eg. when rewriting term to same term plus extra content.
> 
> With new macro we can actually prevent further rewriting on marked expr or
> stmts, eg. with given example echo(...) will be rewritten just once:
> 
>   ```Nim
>   template pwnEcho{echo(x)}(x: expr) =
>     {.noRewrite.}: echo("pwned!")
> 
>   echo "ab"
>   ```

cc @andreaferretti @kaushalmodi @reactormonk

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Fix #9454

### CODE REVIEW

**Comment by Araq** [comment]
This belongs into the manual_experimental.rst

File: doc/manual.rst [hunk 1]
```diff
@@ -6020,6 +6020,27 @@ In the example a new pragma named ``rtl`` is introduced that either imports
 a symbol from a dynamic library or exports the symbol for dynamic library
 generation.
 
+noRewrite pragma
+----------------
+
+Term rewriting macros and templates are currently greedy and
+they will rewrite as long as there is a match.
+So there was no way to ensure some rewrite happens only once,
```
Comment (narimiran) [inline doc/manual.rst]: Remove `So` from the beginning of the sentence.

File: doc/manual.rst [hunk 2]
```diff
@@ -6020,6 +6020,27 @@ In the example a new pragma named ``rtl`` is introduced that either imports
...
+eg. when rewriting term to same term plus extra content.
```
Comment (narimiran) [inline doc/manual.rst]: Change `eg.` to `e.g.` (here and below)

File: doc/manual.rst [hunk 3]
```diff
@@ -6020,6 +6020,27 @@ In the example a new pragma named ``rtl`` is introduced that either imports
...
+
+noRewrite pragma can actually prevent further rewriting on marked code,
```
Comment (narimiran) [inline doc/manual.rst]: Put double backticks around `noRewrite` (here and below), like it was done for other pragmas above.

File: doc/manual.rst [hunk 4]
```diff
@@ -6020,6 +6020,27 @@ In the example a new pragma named ``rtl`` is introduced that either imports
...
+eg. with given example ``echo("ab")`` will be rewritten just once:
+
+Example:
```
Comment (narimiran) [inline doc/manual.rst]: No need for `Example:` since the previous line begins with `e.g.` and finishes with a colon.

**Comment by juancarlospaco** [comment]
@narimiran Check again, is all fixed. :slightly_smiling_face: :+1:

**Comment by Araq** [comment]
Unrelated CI failure, merging.

**Pull request merged**